### DISCUSSION TITLE
category-roslynator.severity not working

### DISCUSSION THREAD

**Discussion by chanakya-svt** [JosefPihrt/Roslynator#875] [General]
I set the following setting in `.editorconfig` file. None of the RCS#### errors are being picked up in VSCode or via CLI `v0.3.1`

```
dotnet_analyzer_diagnostic.category-roslynator.severity = error
roslynator_refactorings.enabled = true
roslynator_compiler_diagnostic_fixes.enabled = true
```

VSCode version: 1.63.2
Dotnet: v5
Roslynator.Analyzers: v4.0.0-rc
Roslynator.CodeAnalysis.Analyzers: v4.0.0-rc
Roslynator.Formatting.Analyzers: v4.0.0-rc
roslynator.dotnet.cli(globally installed): v0.3.1

I used the configuration from [here](https://github.com/JosefPihrt/Roslynator/blob/master/docs/Configuration.md)

Is there anything else I need to setup for this to work or I need to set individual severity for each ANALYZER_ID?

**Comment by josefpihrt** [comment]
Hi,

I've published new version that fixes this bug. Please install latest version.

Thanks for understanding.

**Reply by chanakya-svt** [reply]
Thank you for your quick response. 

I see there is a new version(4.0.1) in [change log](https://github.com/JosefPihrt/Roslynator/blob/master/ChangeLog.md#401-2022-01-21) but I do not see that in the [Nuget version list](https://www.nuget.org/packages/Roslynator.Analyzers)

**Comment by josefpihrt** [comment]
Hi,

I should probably release nugets as 4.0.1, it would be more clear. But because previous version was 4.0.0-rc I did not want to skip 4.0.0.

Regardless of this confusion, the nuget 4.0.0 are newest nugets and they contain the fix.

**Reply by chanakya-svt** [reply]
Cool. Thanks for the clarification.

**Comment by chanakya-svt** [comment]
When I run `roslynator analyze` I do not see any of the RCSxxxx errors being reported. They are being reported by the VSCode Roslynator plugin but not via CLI. Only CAxxxx errors are being reported. 

Versions 
- .NET v6.0.100
- Roslynator.Analyzers v4.0.0
- Roslynator.CodeAnalysis.Analyzers v4.0.0
- Roslynator.Formatting.Analyzers v4.0.0
- Microsoft.CodeAnalysis.CSharp.Workspaces v4.0.1
- Roslynator Dotnet CLI v0.3.1.0

I specifically set some RCS severity to `error` for testing. 

Is there any other configuration that is needed for .net6.0?

**Note: The RCSxxxx errors along with CAxxxx are being reported both by cli and viscose plugin when using .net5**

**Comment by josefpihrt** [comment]
Can you please run `roslynator analyzer -v diag` to see diagnostic output?

**Reply by chanakya-svt** [reply]
Please find the attached file with some redaction. 

```
The "ProcessFrameworkReferences" task failed unexpectedly.
System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. Could not find or load a specific file. (0x80131621)
File name: 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
 ---> System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'.
```

Getting the above error when run with `dotnet roslynator analyze -v diag --msbuild-path /usr/local/share/dotnet/sdk/6.0.101/`

I checked my references and I am using v13.0.1 for `Newtonsoft.Json`. 

[diag.txt](https://github.com/JosefPihrt/Roslynator/files/7928912/diag.txt)

**Comment by josefpihrt** [comment]
That's unfortunate. But I have suspicion that this is not Roslynator error but MSBuild locator error which is a Microsoft library Roslynator relies upon. You can see it in a call stack as `Microsoft.Build.Locator`.

I would suggest to check their [repo](https://github.com/microsoft/MSBuildLocator) and file an issue there.

**Reply by chanakya-svt** [reply]
Hello JosefPihrt,

I want to check in and see if anyone else is facing the issue. I raised the issue in the `Microsoft.Build.Locator` but no reply yet. 

However I did find something recently. In my `.csproj` file if I use `<Project Sdk="Microsoft.NET.Sdk">` everything works as expected. But when I change it to `<Project Sdk="Microsoft.NET.Sdk.Web">` I get the same above error.

here is the sample [repo](https://github.com/chanakya-svt/RoslynatorTest) for your reference.(Currently in csproj its just `Microsoft.NET.Sdk`) 

I am using `.net v6.0.200`, running the command line: `roslynator analyze RoslynatorTest.csproj --verbosity d`

**Comment by josefpihrt** [comment]
Hi @chanakya-svt,

It's good that you raised the issue in the Locator repo. I think we should wait for them to answer before making any further assumptions.
